[Docs] updating information about fetch caching in the fetching docs (#72156)
vercel/next.js: docs/02-app/01-building-your-application/02-data-fetching/01-fetching.mdx

Next.js uses APIs like `generateMetadata` and `generateStaticParams` where you will need to use the same data fetched in the `page`.

If you are using `fetch`, requests can be [memoized](/docs/app/building-your-application/caching#request-memoization) by adding `cache: 'force-cache'`. This means you can safely call the same URL with the same options, and only one request will be made.

> **Good to know:**
>

> > - In previous versions of Next.js, using `fetch` will automatically sets the `cache` to `cache: force-cache`. This changed in version 15, to `cache: no-store`

Reviewer: ```suggestion
> - In previous versions of Next.js, using `fetch` would have a default `cache` value of `force-cache`. This changed in version 15, to a default of `cache: no-store`.
```

Next.js uses APIs like `generateMetadata` and `generateStaticParams` where you will need to use the same data fetched in the `page`.

> If you are using `fetch`, requests are no longer being automatically [memoized](/docs/app/building-your-application/caching#request-memoization). Read the fetch changes [here](/docs/app/building-your-application/upgrading/version-15#fetch-requests)

Reviewer: ```suggestion
If you are using `fetch`, requests can be [memoized](/docs/app/building-your-application/caching#request-memoization) by adding `cache: 'force-cache'`. This means you can safely call the same URL with the same options, and only one request will be made.
```

Maybe we mention how to opt into the caching as it's the point of the example below.

Author: i'll add it to the code example too
